store/tikv/scan.go

```diff
@@ -98,10 +100,6 @@ func (s *Scanner) Next() error {
 			s.Close()
 			return errors.Trace(err)
 		}
-		if len(s.Value()) == 0 {
```

Comment:
seem we need to check keyOnly here?

Reply:
AFAIK, the current situation is that TiKV allows empty values, but TiDB does not because an empty value in the `BufferStore` indicates deletion.

Reply:
> AFAIK, the current situation is that TiKV allows empty values
Yeah, that’s why I said an empty value does not mean that a key is non-exist. The code here is weird, I don’t know what is the original design purpose. Can any one give some information?

Reply:
But you cannot remove it directly here. Be aware that `resolveCurrentLock()` used `snapshot.Get()`, which returns nil when the key does not exist. You have to change the behaviour of `Get` if you really want to accept empty values.

Reply:
I checked the `keyOnly` option and the CI passed. However, I think the problem is not resolved. The `keyOnly` option does not change the behavior of TiKV server, so is there any case that the server returns a key which does not exist?